src/main/java/org/truffleruby/core/module/ModuleNodes.java

```diff
@@ -961,7 +961,7 @@ public Object setConstant(DynamicObject module, String name, Object value) {
             }
 
             final RubyConstant previous = Layouts.MODULE.getFields(module).setConstant(getContext(), this, name, value);
-            if (previous != null) {
+            if (previous != null && !previous.isAutoload()) {
```

Comment:
Would it be possible/better to handle this check in ‘setConstant’?  I’m not sure where is better to handle this check.

Reply:
You are right, this should be handled in `setConstant`, fixed.